src/account/default.ts

```diff
@@ -394,27 +433,40 @@ export class Account extends Provider implements AccountInterface {
 
   public async deployContract(
     payload: UniversalDeployerContractPayload | UniversalDeployerContractPayload[],
-    details?: InvocationsDetails | undefined
+    details?: InvocationsDetails | undefined,
+    ...addsAbstraction: BigNumberish[]
   ): Promise<DeployContractUDCResponse> {
-    const deployTx = await this.deploy(payload, details);
+    const adds: string[] = addsAbstraction.map((param) => toHex(param));
+    const deployTx = await this.deploy(payload, details, ...adds);
     const txReceipt = await this.waitForTransaction(deployTx.transaction_hash);
     return parseUDCEvent(txReceipt as DeployTransactionReceiptResponse);
   }
 
   public async declareAndDeploy(
     payload: DeclareAndDeployContractPayload,
-    details?: InvocationsDetails | undefined
+    details?: InvocationsDetails | undefined,
+    addsAbstractionDeclare?: BigNumberish[],
+    addsAbstractionDeploy?: BigNumberish[]
   ): Promise<DeclareDeployUDCResponse> {
     const { constructorCalldata, salt, unique } = payload;
-    let declare = await this.declareIfNot(payload, details);
+
+    let declare = addsAbstractionDeclare
+      ? await this.declareIfNot(payload, details, ...addsAbstractionDeclare)
+      : await this.declareIfNot(payload, details);
     if (declare.transaction_hash !== '') {
       const tx = await this.waitForTransaction(declare.transaction_hash);
       declare = { ...declare, ...tx };
     }
-    const deploy = await this.deployContract(
-      { classHash: declare.class_hash, salt, unique, constructorCalldata },
-      details
-    );
+    const deploy = addsAbstractionDeploy
```

Comment:
can this ternary operator be  avoided and handled in the `deployContract` function?